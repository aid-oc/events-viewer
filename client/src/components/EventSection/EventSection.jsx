import React, { useState } from 'react';
import { useQuery } from '@apollo/client';
import getAllEvents from '../../graph/queries/getAllEvents';
import EventList from '../EventList/EventList';
import { Section, Search } from './EventSection.styles';

const filterEvents = (filter, events) => events.filter((eventEntry) => {
  const eventData = JSON.stringify(eventEntry.event);
  const posterData = JSON.stringify(eventEntry.poster);
  return !filter || eventData.includes(filter) || posterData.includes(filter);
});

const EventSection = () => {
  const { loading, error, data } = useQuery(getAllEvents);
  const [currentFiltering, setCurrentFiltering] = useState(null);
  return (
    <Section>
      <h2>Event Postings</h2>
      <Search type="search" placeholder="Search..." onChange={(e) => setCurrentFiltering(e.target.value)} />
      { loading && <p>Loading Events...</p>}
      { error && <p>Failure loading events, sorry!</p>}
      { data && <EventList events={filterEvents(currentFiltering, data.events)} />}
    </Section>
  );
};

export default EventSection;
