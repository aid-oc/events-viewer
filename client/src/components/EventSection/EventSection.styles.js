import styled from 'styled-components';

const Section = styled.section`
    text-align: center;
    padding: 1rem;
`;

const Search = styled.input`
    border: 1px solid black;
    border-radius: 0.3rem;
    min-width: 15rem;
`;

export { Section, Search };
