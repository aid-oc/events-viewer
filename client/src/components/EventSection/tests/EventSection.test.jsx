import React from 'react';
import {
  render,
} from '@testing-library/react';
import EventSection from '../EventSection';

jest.mock('@apollo/client', () => ({
  ...jest.requireActual('@apollo/client'),
  useQuery: jest.fn(),
}));
const apolloMock = require('@apollo/client');

describe('<EventSection />', () => {
  it('Should display a loading state when the query has not yet fetched data', () => {
    apolloMock.useQuery.mockImplementation(() => ({ loading: true, error: false, data: null }));
    const { asFragment } = render(
      <EventSection />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should display an error state when the query fails', () => {
    apolloMock.useQuery.mockImplementation(() => ({ loading: false, error: true, data: null }));
    const { asFragment } = render(
      <EventSection />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should display data if the query was successful', () => {
    apolloMock.useQuery.mockImplementation(() => (
      {
        loading: false,
        error: false,
        data: { events: [] },
      }));
    const { asFragment } = render(
      <EventSection />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
