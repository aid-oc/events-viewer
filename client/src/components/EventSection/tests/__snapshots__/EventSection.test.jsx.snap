// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventSection /> Should display a loading state when the query has not yet fetched data 1`] = `
<DocumentFragment>
  <section
    class="sc-AxgMl inBwNf"
  >
    <h2>
      Event Postings
    </h2>
    <input
      class="sc-AxheI gEtOKP"
      placeholder="Search..."
      type="search"
    />
    <div
      class="sc-Axmtr hYuCOA"
    >
      <label
        for="showCancelledEvents"
      >
        <input
          id="showCancelledEvents"
          name="showCancelledEvents"
          type="checkbox"
        />
        Hide cancelled events?
      </label>
    </div>
    <p>
      Loading Events...
    </p>
  </section>
</DocumentFragment>
`;

exports[`<EventSection /> Should display an error state when the query fails 1`] = `
<DocumentFragment>
  <section
    class="sc-AxgMl inBwNf"
  >
    <h2>
      Event Postings
    </h2>
    <input
      class="sc-AxheI gEtOKP"
      placeholder="Search..."
      type="search"
    />
    <div
      class="sc-Axmtr hYuCOA"
    >
      <label
        for="showCancelledEvents"
      >
        <input
          id="showCancelledEvents"
          name="showCancelledEvents"
          type="checkbox"
        />
        Hide cancelled events?
      </label>
    </div>
    <p>
      Failure loading events, sorry!
    </p>
  </section>
</DocumentFragment>
`;

exports[`<EventSection /> Should display data if the query was successful 1`] = `
<DocumentFragment>
  <section
    class="sc-AxgMl inBwNf"
  >
    <h2>
      Event Postings
    </h2>
    <input
      class="sc-AxheI gEtOKP"
      placeholder="Search..."
      type="search"
    />
    <div
      class="sc-Axmtr hYuCOA"
    >
      <label
        for="showCancelledEvents"
      >
        <input
          id="showCancelledEvents"
          name="showCancelledEvents"
          type="checkbox"
        />
        Hide cancelled events?
      </label>
    </div>
    <ul
      class="sc-AxhUy gQZmaM"
    />
  </section>
</DocumentFragment>
`;
